asynction/mock_server.py

```diff
@@ -210,7 +211,9 @@ def from_spec(
         )
 
     def _register_handlers(
-        self, default_error_handler: Optional[ErrorHandler] = None
+        self,
+        default_error_handler: Optional[ErrorHandler] = None,
+        server_security: Optional[Sequence[SecurityRequirement]] = None
```

Comment:
The connection handler(s) of the mock server should also honor the security schemes.
I do understand that this is out of the scope of the initial draft PR, however I'm putting it down as a note for the future.